src/command/supergraph/compose.rs

```diff
@@ -43,3 +61,197 @@ impl Compose {
         }
     }
 }
+
+pub(crate) fn get_subgraph_definitions(
+    supergraph_config: SupergraphConfig,
+    config_path: &Utf8PathBuf,
+    client_config: StudioClientConfig,
+    profile_name: &str,
+) -> Result<Vec<SubgraphDefinition>> {
+    let mut subgraphs = Vec::new();
+
+    for (subgraph_name, subgraph_data) in &supergraph_config.subgraphs {
+        match &subgraph_data.schema {
+            SchemaSource::File { file } => {
+                let relative_schema_path = match config_path.parent() {
+                    Some(parent) => {
+                        let mut schema_path = parent.to_path_buf();
+                        schema_path.push(file);
+                        schema_path
+                    }
+                    None => file.clone(),
+                };
+
+                let schema = fs::read_to_string(&relative_schema_path).map_err(|e| {
+                    let err = anyhow!("Could not read \"{}\": {}", &relative_schema_path, e);
+                    let mut err = RoverError::new(err);
+                    err.set_suggestion(Suggestion::ValidComposeFile);
+                    err
+                })?;
+
+                let url = &subgraph_data.routing_url.clone().ok_or_else(|| {
+                    let err = anyhow!("No routing_url found for schema file.");
+                    let mut err = RoverError::new(err);
+                    err.set_suggestion(Suggestion::ValidComposeRoutingUrl);
+                    err
+                })?;
```

Comment:
This calls for us to probably create a RoverCliError type so we can set these outside our actual commands. I'll do this in a separate PR.